### PR TITLE
Passes scrollPhysics from TextFormField to underlying TextField

### LINKED ISSUE #48326: Mirror scrollPhysics in TextFormField
From an old PR:

> While TextField now has a scrollPhysics property to disable/customize scrolling, TextFormField does not currently expose the property.

This seems like a reasonable change. The PR that first proposed it, #36562, needed tests, and has been closed after a few months without a response. This sounds like a nice change for new contributors. :)

### PR DESCRIPTION
**Pull request by mannprerak2**
## Description

Passes scrollPhysics from TextFormField to parent TextField, also added this test to
packages/flutter/test/material/text_form_field.dart file

## Related Issues
Fixes #48326 

## Tests

I added the following tests:
```dart
testWidgets('Passes scrollPhysics to underlying TextField', (WidgetTester tester) async {
    const ScrollPhysics scrollPhysics = ScrollPhysics();

    await tester.pumpWidget(
      MaterialApp(
        home: Material(
          child: Center(
            child: TextFormField(
              scrollPhysics: scrollPhysics,
            ),
          ),
        ),
      ),
    );
```

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.
- [ ] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Did any tests fail when you ran them? Please read [Handling breaking changes].

- [x] No, no existing tests failed, so this is *not* a breaking change.
- [ ] Yes, this is a breaking change. *If not, delete the remainder of this section.*
   - [ ] I wrote a design doc: https://flutter.dev/go/template *Replace this with a link to your design doc's short link*
   - [ ] I got input from the developer relations team, specifically from: *Replace with the names of who gave advice*
   - [ ] I wrote a migration guide: *Replace with a link to your migration guide*

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Test Coverage]: https://github.com/flutter/flutter/wiki/Test-coverage-for-package%3Aflutter
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

### CODE REVIEW

**Comment by Piinks** [comment]
Hey @mannprerak2, welcome! And thank you for the contribution! 🎉 
This change looks ok to me! I will defer to @hansmuller and @justinmc though.

**Review by justinmc — APPROVED** [review]
This looks good to me, thanks for the contribution!  I should have added this myself in https://github.com/flutter/flutter/pull/31088.

I'll try to merge this when the tree is green.

**Pull request merged**
